### PR TITLE
ci: reduce duplication in integration tests

### PR DESCRIPTION
**Pull request by thomaseizinger**

### CODE REVIEW

**Review by jamilbk — APPROVED** [review]

File: scripts/tests/lib.sh [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 
+set -euo pipefail
```
Comment (ReactorScram) [inline scripts/tests/lib.sh:3]: This forces the top-level script to have `set -euo pipefail` and means you can't source `lib.sh` from an interactive session. The first error will exit the whole Bash session

**Review by ReactorScram — CHANGES_REQUESTED** [review]

File: scripts/tests/lib.sh (hunk 1 above)
Comment (jamilbk) [reply scripts/tests/lib.sh:3]: Ah, I think we've assumed these scripts are just for CI.

Are they intended to be run locally?

Comment (ReactorScram) [reply scripts/tests/lib.sh:3]: Yeah I guess not. If I need it I'll write something else and not check it in

**Review by ReactorScram — APPROVED** [review]
I can't figure out how to delete the other review so I'm making this one 'approve' but I think this is pointless and I don't like it.

File: scripts/tests/lib.sh (hunk 1 above)
Comment (jamilbk) [reply scripts/tests/lib.sh:3]: This doesn't have to go in; I don't feel strongly either way. 

A case could be made against using `set ...` in lib scripts anyhow

**Comment by thomaseizinger** [comment]
> I can't figure out how to delete the other review so I'm making this one 'approve' but I think this is pointless and I don't like it.

I should have added some clarification on why I want this!

Bash is notoriously hard to get right and full of quirks. Unifying how we write our integration tests and making sure every test file sets the right options reduces some of this risk. Originally, I also wanted to add a `trap ERR` that displays the line-number at which a certain test failed but I couldn't make it work so this is all that is left.

**Pull request merged**